packages/jaeger-ui/src/utils/readJsonFile.tsx

```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import JaegerAPI from '../api/jaeger';
+
 export default function readJsonFile(fileList: { file: File }): Promise<string> {
   return new Promise((resolve, reject) => {
```

Comment:
This looks like a generic function that just loads and parses a JSON file. It doesn't have anything to do with specifically loading traces (could be loading a config file). I think your changes need to go in upload-related components which are calling this `readJsonFile` function, and once they receive the JSON and start processing it then the new logic to intercept OTLP should be placed.